client/src/components/RoleDropdown.tsx

```diff
@@ -0,0 +1,27 @@
+import { ReactElement } from "react"
+import { Role } from "../game/roleState.d"
+import React from "react"
+import ROLES from "../resources/roles.json"
+import translate from "../game/lang"
+
+export default function RoleDropdown(props: {
+    value: Role,
+    onChange: (role: Role) => void,
+    disabledRoles?: Role[]
+}): ReactElement {
+
+
+    let options = Object.keys(ROLES)
+        .filter((role)=>!props.disabledRoles?.includes(role as Role))
```

Comment:
After I merge buttons PR there will be a getAllRoles function in roleListState.d.tsx